internal/config/config.go

```diff
@@ -185,32 +179,44 @@ func (c *Configuration) SanitizeAPIUI(ctx context.Context) (err error) {
 	}
 
 	log.Info(ctx, "Checking vault token", "token", c.KeyStore.Token)
-	if c.KeyStore.Token == "" {
-		c.KeyStore.Token, err = loadValueFromFile(ctx, k8sVaultTokenFile, k8NRetries, k8TBetweenRetries)
-		if err != nil {
-			return fmt.Errorf("a vault token must be provided")
-		}
-		log.Info(ctx, "Vault token loaded from file", "token", c.KeyStore.Token)
+	if c.KeyStore.Token == "" && !c.VaultUserPassAuthEnabled {
+		log.Error(ctx, "a vault token must be provided or vault userpass auth must be enabled", "vaultUserPassAuthEnabled", c.VaultUserPassAuthEnabled)
+		return fmt.Errorf("a vault token must be provided or vault userpass auth must be enabled")
 	}
 
-	log.Info(ctx, "Checking issuer did value", "did", c.APIUI.Issuer)
-	if c.APIUI.Issuer == "" {
-		c.APIUI.Issuer, err = loadValueFromFile(ctx, K8sDidFile, k8NRetries, k8TBetweenRetries)
+	if c.APIUI.Issuer != "" {
+		issuerDID, err := core.ParseDID(c.APIUI.Issuer)
 		if err != nil {
-			return fmt.Errorf("an issuer DID must be provided")
+			log.Error(ctx, "invalid issuer did format", "error", err)
+			return fmt.Errorf("invalid issuer did format")
 		}
+		c.APIUI.IssuerDID = *issuerDID
+	} else {
+		log.Info(ctx, "Issuer DID not provided in configuration file")
 	}
 
-	log.Info(ctx, "Issuer Did from file", "did", c.APIUI.Issuer)
+	return nil
+}
 
-	issuerDID, err := core.ParseDID(c.APIUI.Issuer)
-	if err != nil {
-		log.Error(ctx, "invalid issuer did format", "error", err)
-		return fmt.Errorf("invalid issuer did format")
+// CheckDID checks if the issuer did is provided in the configuration file. If not, it tries to get it from vault.
+func CheckDID(ctx context.Context, cfg Configuration, vaultCli *api.Client) error {
+	log.Info(ctx, "Checking issuer did value", "did", cfg.APIUI.Issuer)
+	if cfg.APIUI.Issuer == "" {
```

Comment:
As the IssuerDID can be provided in the config file I would not delete it from the sample, The rest of the PR LGTM, awesome job!

https://github.com/0xPolygonID/issuer-node/pull/474/files#diff-b9c16d00e36e5e4426865988b887442b355f46be62264002d2116ea5bf21f352L7